File: FoodDeliveryApp/FoodDeliverApp-API/order-service/src/main/java/com/fooddelivery/order_service/entity/Order.java
package com.fooddelivery.order_service.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "orders")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long customerId;
    private Long restaurantId;
    private String item;
    private int quantity;
    private String status; // PENDING, DELIVERED, CANCELLED
    private LocalDateTime createdAt;
}

File: FoodDeliveryApp/FoodDelivery-UI/order-app/src/types/Order.ts
export interface Order {
  id: number;
  customerId: number;
  restaurantId: number;
  items: { name: string; quantity: number; price: number }[];
  totalAmount: number;
  status: "PENDING" | "CONFIRMED" | "DELIVERED" | "CANCELLED";
  createdAt: string;
}
